ci/test_wheel_cugraph-dgl.sh

```diff
@@ -30,20 +21,16 @@ else
   PYTORCH_CUDA_VER=$PKG_CUDA_VER
 fi
 PYTORCH_URL="https://download.pytorch.org/whl/cu${PYTORCH_CUDA_VER}"
-DGL_URL="https://data.dgl.ai/wheels/cu${PYTORCH_CUDA_VER}/repo.html"
-
-# Starting from 2.2, PyTorch wheels depend on nvidia-nccl-cuxx>=2.19 wheel and
-# dynamically link to NCCL. RAPIDS CUDA 11 CI images have an older NCCL version that
-# might shadow the newer NCCL required by PyTorch during import (when importing
-# `cupy` before `torch`).
-if [[ "${NCCL_VERSION}" < "2.19" ]]; then
-  PYTORCH_VER="2.1.0"
-else
-  PYTORCH_VER="2.3.0"
-fi
-
-rapids-logger "Installing PyTorch and DGL"
-rapids-retry python -m pip install "torch==${PYTORCH_VER}" --index-url ${PYTORCH_URL}
-rapids-retry python -m pip install dgl==2.0.0 --find-links ${DGL_URL}
-
-python -m pytest python/cugraph-dgl/tests
+DGL_URL="https://data.dgl.ai/wheels/torch-2.3/cu${PYTORCH_CUDA_VER}/repo.html"
+
+# echo to expand wildcard before adding `[extra]` requires for pip
+python -m pip install \
+    -v \
+    --extra-index-url "${PYTORCH_URL}" \
+    --find-links "${DGL_URL}" \
+    "$(echo ./local-deps/pylibwholegraph_${RAPIDS_PY_CUDA_SUFFIX}*.whl)" \
+    "$(echo ./dist/cugraph_dgl_${RAPIDS_PY_CUDA_SUFFIX}*.whl)[test]" \
+    'dgl==2.4.0' \
+    'torch>=2.0,<2.4.0a0'
+
+python -m pytest python/cugraph-dgl/cugraph_dgl/tests
```

Comment:
@alexbarghi-nv @tingyu66 the `cugraph-dgl` tests are failing like this:

```text
ImportError while loading conftest '/__w/cugraph-gnn/cugraph-gnn/python/cugraph-dgl/cugraph_dgl/tests/conftest.py'.
python/cugraph-dgl/cugraph_dgl/__init__.py:18: in <module>
Setting the default backend to "pytorch". You can change it in the ~/.dgl/config.json file or export the DGLBACKEND environment variable.  Valid options are: pytorch, mxnet, tensorflow (all lowercase)
    from cugraph_dgl.graph import Graph
python/cugraph-dgl/cugraph_dgl/graph.py:27: in <module>
    from cugraph_dgl.view import (
E   ImportError: cannot import name 'EmbeddingView' from 'cugraph_dgl.view' (/__w/cugraph-gnn/cugraph-gnn/python/cugraph-dgl/cugraph_dgl/view.py)
```

([build link](https://github.com/rapidsai/cugraph-gnn/actions/runs/11449078487/job/31855217868?pr=59#step:8:2889))

Could you please help by pushing commits here?

Looks to me like that is not a missing dependency issue ... that class genuinely exists in the `cugraph` repo (https://github.com/rapidsai/cugraph/blob/27f8ce1d8c1e4a24e3b15e12fea7a43e5f7566b8/python/cugraph-dgl/cugraph_dgl/view.py#L31) but not here (https://github.com/rapidsai/cugraph-gnn/blob/bf6491413015732afd7296c9f821dcee240b61cb/python/cugraph-dgl/cugraph_dgl/view.py)... but the code here is importing it:

https://github.com/rapidsai/cugraph-gnn/blob/bf6491413015732afd7296c9f821dcee240b61cb/python/cugraph-dgl/cugraph_dgl/graph.py#L32

https://github.com/rapidsai/cugraph-gnn/blob/bf6491413015732afd7296c9f821dcee240b61cb/python/cugraph-dgl/cugraph_dgl/graph.py#L586-L588

Reply:
Oh, I know about this problem.  I have a fix for it.

Reply:
Sorry about that, I've been juggling a lot of things right now and forgot to push the fix.

Reply:
No prob! Take your time, just `@` me whenever this is ready for me to pick up again.

Reply:
@jameslamb I think we can resume this; now the blocker is one of the datasets from `cugraph`.  In my other PR, I added it to the repo, but if there's a better solution (i.e. downloading it) then I'm ok with that.

Reply:
Great, thanks!

I see this in the logs for the `wheel-tests-cugraph-dgl` job:

> RuntimeError: The datafile /github/home/.cugraph/datasets/karate.csv does not exist. Try setting download=True to download the datafile

([build link](https://github.com/rapidsai/cugraph-gnn/actions/runs/11468344691/job/31913591798?pr=59))

Notice I'd also added that file to the repo just as you did in #53: https://github.com/rapidsai/cugraph-gnn/pull/59/files

<img width="587" alt="Screenshot 2024-10-28 at 9 50 39 AM" src="https://github.com/user-attachments/assets/dd6fd3e4-be07-42ef-aed5-b0dcc45593f6">

But that path `/github/home/.cugraph` doesn't look right to me. I'll look into where that's coming from and how to get the tests to find the copy of this file checked into the repo.

Reply:
Found it: https://github.com/rapidsai/cugraph/blob/e892fc27c9d14412a67aaa1171be8eeacd266c6b/python/cugraph/cugraph/datasets/dataset.py#L37

Looks like some testing code in `cugraph` defaults to `${HOME}/.cugraph/datasets`. Something somewhere is not correctly overriding that here.

Reply:
Ah I think I see... `ci/test_wheel_cugraph-dgl.sh` needs to set environment variable `RAPIDS_DATASET_ROOT_DIR` to prevent this testing code from falling back to `${HOME}/.cugraph/dataset`.

Like you did for `ci/test_wheel_cugraph-pyg.sh` over in #53:

https://github.com/alexbarghi-nv/cugraph-gnn/blob/be6fc5a5a501f30371ffa032c529997c3e23f8c4/ci/test_wheel_cugraph-pyg.sh#L39

In local testing, I found that that change was sufficient to get some of the failing tests passing, so pushed https://github.com/rapidsai/cugraph-gnn/pull/59/commits/e718840765d47a971d641576e269c44848e213e7 with that change.

Reply:
ok yeah that worked: https://github.com/rapidsai/cugraph-gnn/actions/runs/11558506205/job/32171484314?pr=59

`cugraph-dgl` wheel tests are now passing 🎉